app/nexus/internal/state/base/path.go

```diff
@@ -22,7 +32,21 @@ package base
 //	    fmt.Printf("Found key: %s\n", key)
 //	}
 func ListKeys() []string {
-	secretStoreMu.Lock()
-	defer secretStoreMu.Unlock()
-	return secretStore.List()
+	ctx := context.Background()
+
+	secrets, err := persist.Backend().LoadAllSecrets(ctx)
+	if err != nil {
+		return []string{}
+	}
```

Comment:
the change was involved; but oddly-enough certain methods become much simpler w/o the added caching layer.